quinn-proto/src/connection/spaces.rs

```diff
@@ -91,9 +91,18 @@ impl PacketSpace {
     //    return an Option but that would need to be handled for all callers.  This could be
     //    worth exploring once we have all the main multipath bits fitted.
     pub(super) fn for_path(&mut self, path: PathId) -> &mut PacketNumberSpace {
-        self.number_spaces
-            .entry(path)
-            .or_insert_with(|| PacketNumberSpace::new_default(self.space_id, path))
+        match self.number_spaces.entry(path) {
```

Comment:
it seems that the proper fix is to handle the TODO from @flub above? given the assumption is now that the `vaccant` option should never be hit?

Reply:
There already is a version of this function that returns an `Option`.
The TODO is actually somewhat outdated, I tried changing it to return a `&PacketNumberSpace`, but it turns out `for_path` is used to do mutations everywhere, and there would be a lot of places that would either need `.unwrap()`s or that would need to be completely refactored to avoid borrowing issues.

I agree this is something that should be done eventually, but not in the way that the TODO suggests, and definitely not something in this PR.